### PR TITLE
gen 3 useful + useless moves update

### PR DESCRIPTION
**Pull request by Rezzo64**
Read here for more info:

https://www.smogon.com/forums/threads/move-category-suggestions.3671530/page-4#post-10301044

### CODE REVIEW

File: play.pokemonshowdown.com/src/battle-dex-search.ts [hunk 1]
```diff
@@ -1588,10 +1609,10 @@ class BattleMoveSearch extends BattleTypedSearch<'move'> {
...
 	] as ID[] as readonly ID[];
 	static readonly BAD_STRONG_MOVES = [
-		'belch', 'burnup', 'crushclaw', 'dragonrush', 'dreameater', 'eggbomb', 'firepledge', 'flyingpress', 'grasspledge', 'hyperbeam', 'hyperfang', 'hyperspacehole', 'jawlock', 'landswrath', 'megakick', 'megapunch', 'mistyexplosion', 'muddywater', 'nightdaze', 'pollenpuff', 'rockclimb', 'selfdestruct', 'shelltrap', 'skyuppercut', 'slam', 'strength', 'submission', 'synchronoise', 'takedown', 'thrash', 'uproar', 'waterpledge',
+		'belch', 'burnup', 'crushclaw', 'dragonrush', 'dreameater', 'eggbomb', 'firepledge', 'flyingpress', 'futuresight', 'grasspledge', 'hyperbeam', 'hyperfang', 'hyperspacehole', 'jawlock', 'landswrath', 'megakick', 'megapunch', 'mistyexplosion', 'muddywater', 'nightdaze', 'pollenpuff', 'rockclimb', 'selfdestruct', 'shelltrap', 'skyuppercut', 'slam', 'strength', 'submission', 'synchronoise', 'takedown', 'thrash', 'uproar', 'waterpledge',
```
Comment (KrisXV) [inline play.pokemonshowdown.com/src/battle-dex-search.ts:1615]: future sight is insanely viable in gens 8+ so i dont know if this is the best idea

Comment (KrisXV) [reply play.pokemonshowdown.com/src/battle-dex-search.ts:1615]: ah nevermind, you added a check for it above, mb

File: play.pokemonshowdown.com/src/battle-dex-search.ts [hunk 2]
```diff
@@ -1493,24 +1498,32 @@ class BattleMoveSearch extends BattleTypedSearch<'move'> {
 			return !moves.includes('highjumpkick') && !moves.includes('axekick');
 		case 'lastresort':
 			return set && set.moves.length < 3;
+		case 'leafblade':
+			return dex.gen < 4;
```
Comment (Zarel) [inline play.pokemonshowdown.com/src/battle-dex-search.ts:1502]: Why did you make Leaf Blade useless in Gen 4 and higher? I assume this was a mistake?

Comment (Rezzo64) [reply play.pokemonshowdown.com/src/battle-dex-search.ts:1502]: Yeah, looks like a mistake by negation, apologies. I cannot remember what I tested at the time, but I'm at least certain that the intent was just to change Leaf Blade's usefulness in gen 3

**Pull request merged**